src/components/include/sl.h

```diff
@@ -321,6 +328,31 @@ sl_timeout_wakeup_expired(cycles_t now)
 	} while (heap_size(sl_timeout_heap()));
 }
 
+static inline int
+sl_thd_activate(struct sl_thd *t, sched_tok_t tok)
+{
+	struct cos_defcompinfo *dci = cos_defcompinfo_curr_get();
+	struct cos_compinfo    *ci  = &dci->ci;
+
+	switch(t->type) {
+	case SL_THD_THD:
+	case SL_THD_AEP:
+	{
+		return cos_defswitch(sl_thd_thdcap(t), t->prio, sl__globals()->timeout_next, tok);	
+	}
+	case SL_THD_AEP_TCAP:
+	{
+		return cos_switch(sl_thd_thdcap(t), sl_thd_tcap(t), t->prio, sl__globals()->timeout_next, sl__globals()->sched_rcv, tok);	
```

Comment:
I'll explain each thread type: 

1. `SL_THD_THD`: This is a thread that is not associated with RCVCAP or TCAP. So scheduler will basically use it's tcap to schedule this thread using `cos_switch()`. This is how we'd implement normal worker threads. 
2. `SL_THD_AEP`: This is a thread that is associated with it's RCVCAP and associates to scheduler TCAP for it's time. This is not a child component thread, you can think of it as a Interrupt thread. So scheduler would switch to it using `cos_defswitch` by using scheduler tcap's time. This is how we'd implement interrupts that don't need temporal isolation from other threads in that scheduling domain and tcap quality properties entirely. This is for implementing `cosXen` and it's variations that we had in our previous paper where all parts of that VM uses a single TCap. 
3. `SL_THD_AEP_TCAP`: This is very much like the one above, but instead of using scheduler's tcap for time, it uses it's own TCap. This is again an interrupt thread so scheduler would switch to it using `cos_switch` and provides time to that AEP through `cos_tcap_transfer`. 
This is how we'd implement AEP that get both its own time from its scheduler and perhaps delegations from other parts of the system. 
4. `SL_THD_COMP`: This is a child component, and uses current scheduler's tcap for it's time. So the current scheduler doesn't have any additional `cos_tcap_transfer` for transfering time to it. Scheduler dispatches to it using `cos_asnd`. I'm not sure if there is a good reason for me to have this nor do I know specific experiments that we'd use this for. I can only say, we'd use this if everything in the system uses one timeslice source including root, child schedulers. 
5. `SL_THD_COMP_TCAP`: This is a child component, that we'd have in most cases with it's own TCAP. Scheduler either uses `cos_tcap_delegate` or a pair of `cos_tcap_transfer` + `cos_asnd` to provide time and dispatch to it. This is how we'd have most implementations of components, also enables distributed time management through Tcaps. 

Here, AEP doesn't mean use `asnd` to switch to it. Scheduler uses `cos_switch` or `cos_defswitch` for the first 3 types and `cos_asnd` for the last 2. 
My idea here is to be able to provide all possible types of threads that can be required by different implementations and different experiments that we'd do. Especially because we'd use `sl` for most scheduling the system. If you think I have types that don't need to be there ex: `SL_THD_COMP` without it's own TCap, I can remove them.